### PR TITLE
Alzheimer's Classification with Perceiver Transformer ADNI dataset 46983820

### PR DESCRIPTION
**Pull request by lorenzopolicar**
Hello,

This is a pull request for my implementation of using a Perceiver Transformer Model to classify Alzheimer's using the ADNI dataset. My model was able to achieve approximately 65% accuracy on the test set.

The pull request contains the following files:

- `README.md`: My report detailing my model implementation and results in markdown.
- `dataset.py`: Python file that loads the AD_NC dataset to get train, validation, and test data loaders.
- `modules.py`: Python file containing the Perceiver Transformer Model for Alzheimer's Disease Classification
- `train.py`: Python file that initialises the model for Alzheimer's classification, trains and validates it on the corresponding data set, logs the data (loss and accuracy) and saves the model to a file.
- `test.py`: Python file that loads the model and evaluates the model accuracy on the test data set.
- `plot.py`: Python file that generates training plots.
- `requirements.txt`: The Project's dependencies.
- `out`: A directory that contains the logs of the training data
- `saved`: A directory that contains model files
- `plots`: A directory that contain training plot graphs.

Looking forward to hearing your feedback,

Lorenzo Policar
46983820

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: 
- Excellent and detailed


Commit messages: Excellent

Code:
- Uses Perciever transformer
- Well commented
- Good design

Functionality/Performance:
- No patient-level split
- Test acc is 65%


General comments:
- Solves the problem well although performance is poor

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, could have used patient level split -1
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good

**Comment by shakes76** [comment]
For merging needs repo README restored. Does not affect grade only merging of PR.

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks lost.

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks lost.
